llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -2454,8 +2471,16 @@ void InnerLoopVectorizer::emitIterationCountCheck(BasicBlock *Bypass) {
         ConstantInt::get(CountTy, cast<IntegerType>(CountTy)->getMask());
     Value *LHS = Builder.CreateSub(MaxUIntTripCount, Count);
 
+    Value *Step = CreateStep();
+    ScalarEvolution &SE = *PSE.getSE();
+    // Check if we can prove that the trip count is >= the step.
+    const SCEV *TripCountSCEV = SE.applyLoopGuards(SE.getSCEV(LHS), OrigLoop);
```

Comment:
Does this have any side-effects, i.e. change the existing IR in any way? If not, everything from lines 2475-2481 are entirely related to the assert. Perhaps wrap it in a `#ifndef NDEBUG`?

Reply:
It shouldn't have any side-effects (it may lead to a SCEV being cached for `LHS`, but that shouldn't impact anything unless there's a bug in SCEV invalidation).

Wrapped everything related in `#ifndef NDEBUG`, thanks!